Cesium.CodeGen/Contexts/ForScope.cs

```diff
@@ -7,22 +7,22 @@
 
 namespace Cesium.CodeGen.Contexts;
 
-internal record ForScope(IEmitScope Parent) : IEmitScope
+internal record ForScope(IEmitScope Parent) : IEmitScope, IDeclarationScope
 {
     public AssemblyContext AssemblyContext => Parent.AssemblyContext;
     public ModuleDefinition Module => Parent.Module;
     public CTypeSystem CTypeSystem => Parent.CTypeSystem;
-    public IReadOnlyDictionary<string, FunctionInfo> Functions => Parent.Functions;
+    public IReadOnlyDictionary<string, FunctionInfo> Functions => ((IDeclarationScope)Parent).Functions;
     public TranslationUnitContext Context => Parent.Context;
     public MethodDefinition Method => Parent.Method;
-    public IReadOnlyDictionary<string, IType> Variables => Parent.Variables; // no declarations for `for` now, so pass parent variables
-    public IReadOnlyDictionary<string, IType> GlobalFields => Parent.GlobalFields;
+    public IReadOnlyDictionary<string, IType> Variables => ((IDeclarationScope)Parent).Variables; // no declarations for `for` now, so pass parent variables
+    public IReadOnlyDictionary<string, IType> GlobalFields => ((IDeclarationScope)Parent).GlobalFields;
```

Comment:
The casts here are alarming. Should this block have two differently-typed parents now?

Reply:
I try to limit amount of changes which I put in single PR. for now, scopes are should implement two interfaces. And I can pinky swear that I get rid of that. but that's potentially additional drag. At least now all developers cannot cross boundaries of syntax analysis and emit.

If I pass tow scopes for `ForScope` I just move that cast to [different location](https://github.com/ForNeVeR/Cesium/blob/3e07a3bf96f3fe24596af45d75fbcf0a741d3d54/Cesium.CodeGen/Ir/BlockItems/ForStatement.cs#L50). I do not see point to have `ForEmitScope` at all, but current implementation is relying on that, so I have to keep it for sure. but `ForDeclarationScope` is pointless since it was never actually used. I prefer leave this, knowing this is very problematic place which should be solved.

Reply:
Also, this chain of refactoring made me realize that maybe `IEmitScope` and `TranslationUnitContext` is same thing. Not yet sure, but both things look pretty close in spirit to each other.

Reply:
> If I pass tow scopes for ForScope I just move that cast to [different location](https://github.com/ForNeVeR/Cesium/blob/3e07a3bf96f3fe24596af45d75fbcf0a741d3d54/Cesium.CodeGen/Ir/BlockItems/ForStatement.cs#L50).

Ok, this is fair. Let it be, then.